src/library/scala/collection/Seq.scala

```diff
@@ -266,15 +266,17 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     *  @return `true` if this $coll has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = {
-    val i = iterator.drop(length - that.size)
-    val j = that.iterator
-    while (i.hasNext && j.hasNext)
-      if (i.next() != j.next())
-        return false
+    if (this.sizeCompare(that) < 0) false else {
+      val i = iterator.drop(length - that.size)
```

Comment:
does `Iterator` have a `takeRight`? that would potentially avoid computing `length`.